Use GNU diff instead of Solaris diff.

Solaris diff doesn't understand the -N option, which is part of the default
option set.

This is a very Solaris-specific patch, and should not be passed upstream.

--- mercurial-7.1.2/hgext/extdiff.py
+++ mercurial-7.1.2/hgext/extdiff.py
@@ -703,7 +703,7 @@ def extdiff(ui, repo, *pats, **opts):
     program = opts.get(b'program')
     option = opts.get(b'option')
     if not program:
-        program = b'diff'
+        program = b'/usr/gnu/bin/diff'
         option = option or [b'-Npru']
     cmdline = b' '.join(map(procutil.shellquote, [program] + option))
     return dodiff(ui, repo, cmdline, pats, opts)
